corpus/stats.py

```diff
@@ -99,3 +102,48 @@ def run(self):
         counts = [(v, k) for k, v in words.items()]
         with uopen(self.out_word_counts, "wt") as f:
             f.write("\n".join("%d\t%s" % t for t in sorted(counts, key=lambda t: (-t[0], t[1]))))
+
+
+class DumpRecordingAudiosJob(Job):
+    """
+    Dump all recordings of a given corpus file, one audio per line.
+    """
+
+    def __init__(
+        self, corpus_files: Union[tk.Path, List[tk.Path]], dump_durations: bool = False, zip_output: bool = False
+    ):
+        r"""
+        :param corpus_file: Corpus file from which to obtain the audio list.
+        :param dump_durations: Whether to dump the durations of the audios along with the audio list.
+            The `out_audio_durations` output file will contain an audio/duration pair, separated by `\t`.
+        :param zip_output: Whether the output should be zipped.
+        """
+        self.corpus_files = [corpus_files] if isinstance(corpus_files, tk.Path) else corpus_files
+        self.dump_durations = dump_durations
+
+        suffix = ".gz" if zip_output else ""
+        self.out_audio_list = self.output_path(f"out.txt{suffix}")
+        if dump_durations:
+            self.out_audio_durations = self.output_path(f"out_durations.txt{suffix}")
+
+        self.rqmt = {"cpu": 1, "mem": 1.0, "time": 1.0}
+
+    def tasks(self):
+        yield Task("run", resume="run", rqmt=self.rqmt)
+
+    def run(self):
+        if self.dump_durations:
+            f_dur = uopen(self.out_audio_durations.get_path(), "wt")
+
+        with uopen(self.out_audio_list.get_path(), "wt") as f:
+            for corpus_file in self.corpus_files:
+                c = libcorpus.Corpus()
+                c.load(corpus_file.get_path())
+
+                for r in c.all_recordings():
+                    f.write(f"{r.audio}\n")
+                    if self.dump_durations:
+                        f_dur.write(f"{r.audio}\t{compute_rec_duration(r.audio)}\n")
+
+        if self.dump_durations:
+            f_dur.close()
```

Comment:
> Manually open/close file

ah.. I don't find that elegant. let me instead suggest
```suggestion
        with uopen(self.out_audio_list.get_path(), "wt") as f, uopen(self.out_audio_durations.get_path(), "wt") if self.dump_durations else nullcontext() as f_dur:
            for corpus_file in self.corpus_files:
                c = libcorpus.Corpus()
                c.load(corpus_file.get_path())

                for r in c.all_recordings():
                    f.write(f"{r.audio}\n")
                    if self.dump_durations:
                        f_dur.write(f"{r.audio}\t{compute_rec_duration(r.audio)}\n")
```
add `from contextlib import nullcontext` plus let black do its magic.

(kudos to [stackoverflow](https://stackoverflow.com/questions/27803059/conditional-with-statement-in-python))

Reply:
That's wonderful! Thank you for the suggestion! :star_struck: